lib/elixir/lib/kernel/typespec.ex

```diff
@@ -1216,4 +1216,27 @@ defmodule Kernel.Typespec do
   defp unpack_typespec_kw(_, _acc) do
     :error
   end
+
+  # Called by elixir_erl.
+  def spec_related_info(table) do
```

Comment:
Beautiful implementation!

Although the name is too vague?

What about `translate_typespecs` or `translate_typespecs_for_module`?